approval_polls/views.py

```diff
@@ -1,101 +1,212 @@
-  except PageNotAnInteger:
-      polls = paginator.page(1)
-  except EmptyPage:
-      polls = paginator.page(paginator.num_pages)
-  return render(request, render_page, {'latest_poll_list' : polls})
+    paginator = Paginator(poll_list, 5)
+    page = request.GET.get('page')
+    try:
+        polls = paginator.page(page)
+    except PageNotAnInteger:
+        polls = paginator.page(1)
+    except EmptyPage:
+        polls = paginator.page(paginator.num_pages)
+    return render(request, render_page, {'latest_poll_list': polls})
+
 
 class DetailView(generic.DetailView):
-  model = Poll
-  template_name = 'approval_polls/detail.html'
-  def get_queryset(self):
-      return Poll.objects.filter(pub_date__lte=timezone.now())
+    model = Poll
+    template_name = 'approval_polls/detail.html'
+
+    def get_queryset(self):
+        return Poll.objects.filter(pub_date__lte=timezone.now())
+
 
 class ResultsView(generic.DetailView):
-  model = Poll
-  template_name = 'approval_polls/results.html'
-  def get_queryset(self):
-      return Poll.objects.filter(pub_date__lte=timezone.now())
+    model = Poll
+    template_name = 'approval_polls/results.html'
+
+    def get_queryset(self):
+        return Poll.objects.filter(pub_date__lte=timezone.now())
+
 
 @require_http_methods(['POST'])
 def vote(request, poll_id):
-  poll = get_object_or_404(Poll, pk=poll_id)
-  ballot = poll.ballot_set.create(timestamp=timezone.now())
+    poll = get_object_or_404(Poll, pk=poll_id)
+    ballot = poll.ballot_set.create(timestamp=timezone.now())
+
+    for counter, choice in enumerate(poll.choice_set.all()):
+        if 'choice' + str(counter + 1) in request.POST:
+            ballot.vote_set.create(choice=choice)
+            ballot.save()
 
-  for counter, choice in enumerate(poll.choice_set.all()):
-    if 'choice' + str(counter + 1) in request.POST:
-      ballot.vote_set.create(choice = choice)
-      ballot.save()
+    poll.save()
 
-  poll.save()
+    return HttpResponseRedirect(reverse('approval_polls:results', args=(poll.id,)))
 
-  return HttpResponseRedirect(reverse('approval_polls:results', args=(poll.id,)))
 
 def embed_instructions(request, poll_id):
-  link = request.build_absolute_uri('/approval_polls/{}'.format(poll_id))
-  return render(request, 'approval_polls/embed_instructions.html', {'link': link})
+    link = request.build_absolute_uri('/approval_polls/{}'.format(poll_id))
+    return render(request, 'approval_polls/embed_instructions.html', {'link': link})
 
-class CreateView(generic.View):
 
-  @method_decorator(login_required)
-  def get(self, request, *args, **kwargs):
-    return render(request, 'approval_polls/create.html')
+class PollUpdate(generic.View):
+    '''A generic update view for our poll.
+
+    Since this is actually a compound form updating two different models
+    a Poll and "n" number of Choice instances we cannot use the straight forward
+    UpdateView.
+    '''
+
+
+    success_url = '/approval_polls/my-polls'
```

Comment:
Ahhh good point so let me understand this:

"As a poll owner when I successfully edit a poll I want to be returned to the edit page which shows me a success message so that I know I have successfully updated my poll and I can visually verify the changes I just made"